src/client/utility/calculations.js

```diff
@@ -64,6 +64,28 @@ export function casteljauPoint(curve, t) {
     return new THREE.Vector3(Px, Py, Pz);
 }
 
+// 2D implementation of casteljau's algorithm
+export function casteljauPoint2D(curve, t) {
```

Comment:
I'm not sure that this is necessary, can't you just use the 3d version and disregard the z?

Reply:
The 3d version only works with the original json layout of the boat coordinates and so needs "start", "end", "startControl", and "endControl" as part of that structure. I was trying to use it on the structure I made for the 2d visuals, which don't have those key words or the z coordinate to pass to the function.